src/app/[locale]/posts/(components)/PostsPage.tsx

```diff
@@ -317,26 +316,39 @@ interface Block {
   type?: string;
 }
 
-function CustomHits({ categories }: Pick<Props, "categories">) {
+function CustomHits({ categories, params }: Pick<Props, "categories" | "params">) {
+  const [blogPostsSettings, setBlogPostsSettings] = useState<any>();
+  useEffect(() => {
+    async function fetchData() {
+      const response = await fetch(`/api/blogPosts?locale=${params.locale}`);
+      const data = await response.json();
+      setBlogPostsSettings(data);
+    }
+    fetchData();
+  }, []);
```

Comment:
we have not used this kind of fetching logic on any other data why is this different?